src/ServerScriptService/GravityController/GravityController/Collider.lua

```diff
@@ -124,10 +122,33 @@ function create(self, controller)
 end
 
 function init(self)
+	self._maid:Mark(self.FloorDetector.Touched:Connect(function(part)
+		if not part:IsDescendantOf(self.Controller.Character) and part.CanCollide then
+			table.insert(self._floorTouchingParts, part)
+		end
+	end))
+	self._maid:Mark(self.FloorDetector.TouchEnded:Connect(function(part)
+		if not part:IsDescendantOf(self.Controller.Character) and part.CanCollide then
+			wait(0.01)	-- slight delay before removing from cache
+			if not self._floorTouchingParts then
+				return;
+			end
+			for index, value in pairs(self._floorTouchingParts) do
+				if value == part then
```

Comment:
I think this would be a lot more efficient/faster if we just used a dictionary for this:

```Lua
self._floorTouchingParts[part] = true -- when the part is touched
self._floorTouchingParts[part] = nil -- when the part is touch ended
```

Reply:
If I were to guess the reason why you didn't do this is so you could check the length of the array? See this comment for an alternative that will work w/ dictionaries

https://github.com/EgoMoose/Rbx-Gravity-Controller/pull/2/commits/fc8e20105cf0b02a02834a398033375f938a9dfb#diff-5f6aaf7221bb6cd77b091725263e5746f7ddecd3be0c700e882f65bb65ec6693R168

Reply:
> I think this would be a lot more efficient/faster if we just used a dictionary for this:
> 
> ```lua
> self._floorTouchingParts[part] = true -- when the part is touched
> self._floorTouchingParts[part] = nil -- when the part is touch ended
> ```

I like this technique and I'm going to apply it in my local copy as well.